lib/src/conflicts.rs

```diff
@@ -132,15 +132,39 @@ pub fn materialize_conflict(
     conflict: &Conflict,
     output: &mut dyn Write,
 ) -> std::io::Result<()> {
+    match extract_file_conflict_data(store, path, conflict) {
+        None => {
+            // Unless all parts are regular files, we can't do much better than to try to
+            // describe the conflict.
+            describe_conflict(conflict, output)?;
+        }
+        Some(content) => {
+            let removed_slices = content.removes.iter().map(Vec::as_slice).collect_vec();
+            let added_slices = content.adds.iter().map(Vec::as_slice).collect_vec();
+            materialize_merge_result(files::merge(&removed_slices, &added_slices), output)?;
+        }
+    };
+
+    Ok(())
```

Comment:
nit: this line can be removed by dropping the final `?;` in each of the two `match` arms